examples/counter/shared/src/app/mod.rs

```diff
@@ -0,0 +1,172 @@
+use crux_core::render::Render;
+use crux_http::Http;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+const API_URL: &str = "https://crux-counter.fly.dev";
+
+#[derive(Default)]
+pub struct App;
+
+impl crux_core::App for App {
+    type Model = Model;
+    type Event = Event;
+    type ViewModel = ViewModel;
+    type Capabilities = Capabilities;
+
+    fn update(&self, msg: Self::Event, model: &mut Self::Model, caps: &Self::Capabilities) {
+        match msg {
+            Event::Get => {
+                caps.http.get_json(Url::parse(API_URL).unwrap(), Event::Set);
+            }
+            Event::Set(counter) => {
+                model.count = counter.value;
+                caps.render.render();
+            }
+            Event::Increment => {
+                let base = Url::parse(API_URL).unwrap();
+                let url = base.join("/inc");
+                caps.http.post(url.unwrap(), Event::Set)
+            }
+            Event::Decrement => {
+                let base = Url::parse(API_URL).unwrap();
+                let url = base.join("/dec");
+                caps.http.post(url.unwrap(), Event::Set)
+            }
+        }
+    }
+
+    fn view(&self, model: &Self::Model) -> Self::ViewModel {
+        model.into()
+    }
+}
+
+#[derive(Default)]
+pub struct Model {
+    count: isize,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ViewModel {
+    pub count: isize,
```

Comment:
This could be an already formatted string, to make the distinction between model and view model a bit clerer?